### PR TITLE
fix: prefer top level distinct id

### PR DESCRIPTION
**Pull request by pauldambra**
We've a couple of times tripped over relying on distinct id being in properties.

It's validated on the event anyways, so let's use that

### CODE REVIEW

File: posthog/session_recordings/session_recording_helpers.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def preprocess_replay_events(
 
     size_with_headroom = max_size_bytes * 0.95  # Leave 5% headroom
 
-    distinct_id = events[0]["properties"]["distinct_id"]
+    distinct_id = events[0]["distinct_id"]
```
Comment (marandaneto) [inline posthog/session_recordings/session_recording_helpers.py:124]:
```suggestion
    distinct_id = events[0]["properties"].get("distinct_id") or events[0]["distinct_id"]
```

So it's back-compatible at least?

Comment (marandaneto) [reply posthog/session_recordings/session_recording_helpers.py:124]: Not sure if `or` works here, don't use Python that often =D

Comment (pauldambra) [reply posthog/session_recordings/session_recording_helpers.py:124]: you can `distinct_id = events[0]["properties"].get("distinct_id", events[0]["distinct_id"])`

but the events have all along had the distinct id at the top level so we don't need backwards compatibility here

**Review by benjackwhite — APPROVED** [review]
If you are sure 👍

File: posthog/session_recordings/session_recording_helpers.py (hunk 1 above)
Comment (marandaneto) [reply posthog/session_recordings/session_recording_helpers.py]: I know that old events from the old Android SDK don't send as top-level property, if at this point the event is already normalized (with back-filling high-level property), then all good.

Comment (pauldambra) [reply posthog/session_recordings/session_recording_helpers.py]: > I know that old events from the old Android SDK don't send as top-level property,
🤯 

I wouldn't have thought that would get through the pipeline... OK, I'm going to write tests for all the possible combinations explicitly!

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
@pauldambra CI is still unhappy, at least one of the broken tests is related.

**Comment by marandaneto** [comment]
@pauldambra I reopened the issue, I guess just CI has to be fixed, and LGTM?

**Pull request closed without merging**